_ja/overviews/scala3-migration/compatibility-classpath.md

```diff
@@ -0,0 +1,144 @@
+---
+title: クラスパス互換性
+type: section
+description: このセクションでは、Scala 2.13とScala 3クラスファイルの互換性について説明しています。
+num: 3
+previous-page: compatibility-source
+next-page: compatibility-runtime
+language: ja
+---
+
+コード上で、パブリックな型と構文を利用でき、そしてパブリックメソッドは異なるモジュールやライブラリで定義されています。
+そしてできる限り長く型チェックとして働いて、コンパイルフェーズにてコードの意味的な構成を検証し、クラスファイルに含まれている型や構文やメソッドの重要性を読むことができます。
+
+Scala 2ではシグネチャはpickleフォーマットで保存されます。
+Scala 3では少し違っていて、なぜならシグネチャレイアウトよりも多くのTASTyフォーマットに寄り添っているからです。
+しかしながらScala 2.13からScala 3へ移行する目的においては、シグネチャだけでも有効です。
+
+## Scala 3 Unpickler
+
+はじめの良い知らせとしてScala 3のコンパイラはScala 2.13のpickleフォーマットを読み込むことができるということ、すなわち、Scala 2.13によりコンパイルされたモジュールやライブラリに依存したコードの型チェックができるということです。
+
+Scala3のUnpicklerは数年間広範囲的にコミュニティビルドによってテストされています。なので使うこと自体は安全です。
+
+### Scala 3モジュールはScala 2.13アーティファクトに依存できます
+
+![Scala 3 module depending on a Scala 2.13 artifact](/resources/images/scala3-migration/compatibility-3-to-213.svg)
+
+sbtビルドとして、次のように説明できます（sbt 1.5.0以降が必要です）:
+
+```scala
+lazy val foo = project.in(file("foo"))
+  .settings(scalaVersion := "3.0.0")
+  .dependsOn(bar)
+
+lazy val bar = project.in(file("bar"))
+  .settings(scalaVersion := "2.13.6")
+```
+
+または、barが公開されたScala 2.13ライブラリである場合は、次のことができます。:
+
+```scala
+lazy val foo = project.in(file("foo"))
+  .settings(
+    scalaVersion := "3.0.0",
+    libraryDependencies += ("org.bar" %% "bar" % "1.0.0").cross(CrossVersion.for3Use2_13)
+  )
+```
+
+sbtで `CrossVersion.for3Use2_13`を使用して、 `bar_3`の代わりに`bar_2.13`を解決します。
+
+### スタンダードライブラリ
+
+1つ目のノート例はScala 2.13ライブラリです。
+Scala 2.13ライブラリがScala 3用の公式スタンダードライブラリとして確かに決定しています。
+
+スタンダードライブラリは自動的にビルドツールに与えられることを覚えておきましょう、なので手動的に設定する必要はありません。
+
+## Scala 2.13のTASTy読み取り機能
+
+2つ目の良い知らせはscala 2.13.4からリリースされたTASTy読み取り機能がScala 3ライブラリで使えるということです。
```

Comment:
```suggestion
2つ目の良い知らせは、Scala 2.13.4 より Scala 2.13 TASTy reader がリリースされ、Scala 2.13 側から Scala 3 で書かれたライブラリ群の呼び出しが可能となったことだ。
```

原文: The second piece of good news is that the Scala 2.13 TASTy reader, which enables consuming Scala 3 libraries has been shipped since Scala 2.13.4.

- Scala 3 libraries と複数形になっているのは、標準ライブラリと別なんだよと日本語で区別するために「Scala 3 で書かれたライブラリ群」と訳しました。